GitPrompt.ps1

```diff
@@ -105,12 +105,66 @@ if (Get-Module NuGet) {
     $WindowTitleSupported = $false
 }
 
+# Color codes from https://msdn.microsoft.com/en-us/library/windows/desktop/mt638032(v=vs.85).aspx
+function Get-ForegroundVirtualTerminalSequence($Color) {
+    $e = [char]27 + "["
+    switch ($Color) {
+      ([ConsoleColor]::Black)       { "${e}30m" }

```

Comment:
You should use `break`, otherwise every case is needlessly tested.
You should also consider ordering the cases so that the more commonly used colors appear first - PowerShell does not generate efficient switch tables so each case is tested before the preceding cases.